src/types/chat.d.ts

```diff
@@ -0,0 +1,106 @@
+export interface ChatRoom {
+  id?: string;
+  user: { id: number; participation: boolean };
+  lecturer: { id: number; participation: boolean };
+  roomId?: string;
+  unreadCount?: number;
+  lastChat?: {
+    chatRoomId: string;
+    id: string;
+    sender:
+      | {
+          userId: number;
+          lecturerId: null;
+        }
+      | {
+          userId: null;
+          lecturerId: number;
+        };
```

Comment:
```suggestion
        {
          userId: number | null;
          lecturerId: number | null;
        }
```
이렇게 하는 건 어떠신가요오👀 (`receiver`도..)

Reply:
상기 답글처럼 타입 더 엄격하게 진행을 위함이요오